installer/datafiles/base_omsagent.data

```diff
@@ -180,6 +180,19 @@ fi
 # The real conf directory has to be owned by omsagent to be readable by the omsagent daemon
 chown -R omsagent:omsagent ${{CONF_DIR}}
 
+# Should run before onboarding script(Postinstall_300) and enabling OMS service(%Postinstall_500)
+%Postinstall_299
+InstallMySQL2IntoRuby () {
+    echo "----- Installing MySQL ruby gem -----"
+	if ! ${{OMSHOME}}/ruby/bin/gem install mysql2 2>/dev/null ; then
+	  echo "MySQL ruby gem was not installed. MySQL ruby plugin would be disabled."
+	  echo "Please check whether libmysqlclient dependency package is installed."
+	  echo "You may need to 'apt-get install libmysqlclient-dev' or 'yum install mysql-devel',"
+	  echo "then run '${{OMSHOME}}/ruby/bin/gem install mysql2' after omsagent installation to enable MySQL ruby plugin."
+	fi
+}
+InstallMySQL2IntoRuby
```

Comment:
For this sort of thing, you don't need to define it in a subroutine (unless it's being called elsewhere). Delete lines 185 and 193-194, then change indentation.

I do like your comment at line 183, we should have done more of this!

You might want to consider changing 184 to _275 or something, as that gives you an option to install something else before or after this without touching line 184 again.